include/envoy/stats/symbol_table.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <vector>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Stats {
+/**
+ * Interface for shortening and retrieving stat names.
+ * Guarantees that x = decode(encode(x)) for any x.
+ *
+ * Even though the symbol table does manual reference counting, curr_counter_ is monotonically
+ * increasing. So encoding "foo", freeing the sole stat containing "foo", and then re-encoding "foo"
+ * will produce a different symbol each time.
+ */
+class SymbolTable {
+public:
+  typedef uint32_t Symbol;
```

Comment:
Nit: Some slight preference for `using` over `typedef` here.

Reply:
Done.